stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/Compress.java

```diff
@@ -98,21 +98,30 @@ private static void addEntry(ZipOutputStream zos, Path filePath, String fileStr)
      * @throws IOException exception if an error occurrs when compressing
      */
     static OutputStream compressFiles(Path dir, OutputStream outputStream) throws IOException {
-        Stream<Path> list = Files.walk(dir);
         ZipOutputStream zos = new ZipOutputStream(outputStream);
-        list.forEach(p -> {
-            StringJoiner joiner = new StringJoiner("/");
-            for (Path path : dir.relativize(p)) {
-                joiner.add(path.toString());
+        if (Files.isRegularFile(dir)) {
+            Path fileName = dir.getFileName() != null ? dir.getFileName() : null;
+            if (fileName != null) {
+                addEntry(zos, dir, fileName.toString());
+            } else {
+                throw new BLangRuntimeException("Error occurred when compressing");
             }
-            if (Files.isRegularFile(p)) {
-                try {
-                    addEntry(zos, p, joiner.toString());
-                } catch (IOException e) {
-                    throw new BLangRuntimeException("Error occurred when compressing");
+        } else {
+            Stream<Path> list = Files.walk(dir);
+            list.forEach(p -> {
+                StringJoiner joiner = new StringJoiner("/");
+                for (Path path : dir.relativize(p)) {
```

Comment:
What's the use of a for-each here? `relativize()` only returns a `Path` object.

Reply:
Path object is iterable